sklearn/ensemble/_forest.py

```diff
@@ -679,6 +681,36 @@ def feature_importances_(self):
         all_importances = np.mean(all_importances, axis=0, dtype=np.float64)
         return all_importances / np.sum(all_importances)
 
+    def _get_estimators_indices(self):
+        # Get drawn indices along both sample and feature axes
+        for tree in self.estimators_:
+            if not self.bootstrap:
+                yield np.arange(self._n_samples, dtype=np.int32)
+            else:
+                seed = tree.random_state
+                # Operations accessing random_state must be performed identically
+                # to those in `_parallel_build_trees()`
+                sample_indices = _generate_sample_indices(
+                    seed, self._n_samples, self._n_samples_bootstrap
+                )
+
+                yield sample_indices
+
+    @property
+    def estimators_samples_(self):
```

Comment:
Maybe the following name would be more explicit?

```suggestion
    def training_sample_indices_(self):
```

I want the name to convey:

- those are integer indices rather than the samples themselves,
- those are indices to index the training set.

Reply:
Actually we might want to keep the name `estimators_samples_` used in `Bagging*` models for the sake of consistency.

Reply:
I will say that `training_sample_indices_` is a much more descriptive name in my head, and if you think it warrants a deprecation in both Bagging and here, then I'm happy to add that in a later PR.